File: python/test/test_for_each.py
import ThrustRTC as trtc



printf_functor =  trtc.Functor( {}, ['x'],
'''
         printf("%d\\n", x);
''')

darr = trtc.device_vector_from_list([1,2,3,1,2], 'int32_t')
trtc.For_Each(darr, printf_functor)

File: java/test/test_for_each.java
import JThrustRTC.*;

public class test_for_each
{
	public static void main(String[] args) 
	{
        Functor printf_functor = new Functor(new String[]{ "x" }, "        printf(\"%d\\n\", x);\n");
        DVVector vec = new DVVector(new int[] { 1, 2, 3, 1, 2 });
        TRTC.For_Each(vec, printf_functor);		
	}	

}
